### PR TITLE
Clickhouse cdc

### PR DESCRIPTION
**Pull request by pankaj-peerdb**

### CODE REVIEW

File: flow/connectors/clickhouse/cdc.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+type MirrorJobRow struct {
+	MirrorJobName    string
+	Offset           int
+	SyncBatchID      int
+	NormalizeBatchID int
+}
+
+func (c *ClickhouseConnector) getMirrorRowByJobNAme(jobName string) (*MirrorJobRow, error) {
```
Comment (serprex) [inline flow/connectors/clickhouse/cdc.go:48]:
```suggestion
func (c *ClickhouseConnector) getMirrorRowByJobName(jobName string) (*MirrorJobRow, error) {
```

File: flow/connectors/clickhouse/cdc.go [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+	fmt.Printf("\n ******************************* in cdc.go in SyncRecords config: %+v", c.config.S3Integration)
+	//c.config.S3Integration = "s3://avro-clickhouse"
+	rawTableName := c.getRawTableName(req.FlowJobName)
+	c.logger.Info(fmt.Sprintf("pushing records to Clickhouse table %s", rawTableName))
+
+	syncBatchID, err := c.GetLastSyncBatchID(req.FlowJobName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get previous syncBatchID: %w", err)
+	}
+	syncBatchID += 1
```
Comment (serprex) [inline flow/connectors/clickhouse/cdc.go:175]:
```suggestion
```
Now passed as `req.SyncBatchId` after #1131

File: docker-compose-dev.yml [hunk 3]
```diff
@@ -215,4 +215,4 @@ services:
       - flow-api
 
 volumes:
-  pgdata:
+  pgdata:
```
Comment (serprex) [inline docker-compose-dev.yml:218]: why removing newline? use https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

**Review by iskakaushik — APPROVED** [review]

**Comment by pankaj-peerdb** [comment]
tested locally, works fine

**Pull request merged**